### PR TITLE
IATE DLC

### PR DESCRIPTION
**Pull request by felizdavidad**
- IATE introduction to MARIGOLD
- COV terms set up (COV models not implemented yet)
- calc_dpdz() implementation of Kim's minor loss modification

### CODE REVIEW

File: src/MARIGOLD/iate.py [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+    # Instead, we probably would want to have an input switch indicating which mode of IATE we're executing
+
+    ############################################################################################################################
+    #                                                                                                                          #
+    #                                                       CONSTANTS                                                          #
+    #                                                                                                                          #
+    ############################################################################################################################
+    # expcond.m
+
+    p_atm           = 101325                                    # Ambient pressure [Pa]
```
Comment (ajdix) [inline src/MARIGOLD/iate.py:46]: Let's just put the properties in the Condition __init__, optional argument for phases. Then you can change all these to cond.value

File: src/MARIGOLD/iate.py [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+    R_spec          = 287.058                                   # Specific gas constant for dry air [J/kg-K]
+    T               = 293.15                                    # Ambient absolute temperature [K], for calculating air density as a function of pressure along channel
+    
+    ############################################################################################################################
+    #                                                                                                                          #
+    #                                                      COEFFICIENTS                                                        #
+    #                                                                                                                          #
+    ############################################################################################################################
+    # coeff.m
+
```
Comment (ajdix) [inline src/MARIGOLD/iate.py:63]: Are these constants going to be user inputs? You could have them be set on a method basis or directly

File: src/MARIGOLD/iate.py [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+    #                                                                                                                          #
+    #                                                       COVARIANCE                                                         #
+    #                                                                                                                          #
+    ############################################################################################################################
+    # COV.m
+
+    # Yadav
+    # Temporary, Yadav implemented as a bunch of arrays. There must be a better way to do this.
+    # calc_void_cov in Condition.py may be useful
+    if theta == 90 and elbow == False:
```
Comment (ajdix) [inline src/MARIGOLD/iate.py:113]: Drew had his method, that I doubt even made it out of Excel. We probably should implement that

File: src/MARIGOLD/iate.py [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+        ai[0]       = cond.area_avg("ai")                       # [1/m]
+        alpha[0]    = cond.area_avg("alpha")                    # [-]
+        Db[0]       = cond.void_area_avg("Dsm1") / 1000         # [m]
+    
+    # Pressure drop [Pa/m]
+    dpdz            = cond.calc_dpdz(method=dpdz_method, LM_C=LM_C, k_m=k_m, L=(LoverD_restriction*Dh))
+    
+    jf              = cond.jf                                   # [m/s]
+    jgloc           = cond.jgloc                                # [m/s]
+
```
Comment (ajdix) [inline src/MARIGOLD/iate.py:175]: I think jgatm is guaranteed to be present?

File: src/MARIGOLD/iate.py [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+            err = (ure1 - ur) / ur
+
+            ur = ure1
+        ReD = rho_f * ur * Db[i] * (1 - alpha[i]) / mu_f        # Update bubble Reynolds number
+        CDwe = 24 * (1 + 0.1 * ReD**0.75) / ReD                 # Update drag coefficient
+        '''
+
+        # Original DOE_MATLAB_IAC
+        ReD = rho_f * ur * Db[i] * (1 - alpha[i]) / mu_f
+        CDwe = 24 * (1 + 0.1 * ReD**0.75) / ReD
```
Comment (ajdix) [inline src/MARIGOLD/iate.py:228]: Could use cond.calc_CD method here

Comment (ajdix) [reply src/MARIGOLD/iate.py:228]: That way it'll interface with ur and such

File: src/MARIGOLD/iate.py [hunk 6]
```diff
@@ -0,0 +1,364 @@
...
+        # Estimate Void Fraction for the next step calculation
+
+        if void_method == 'driftflux':      # Drift Flux Model
+
+            j = jgloc + jf
+            
+            # Drift Velocity
+            # Applicable for void fractions less than 20%; for void fractions greater than 30%, use Kataoka and Ishii 1987 for drift-velocity
+            vgj = (2**0.5) * (sigma * grav * (rho_f - rho_gz[i]) / (rho_f**2))**(0.25)
+
```
Comment (ajdix) [inline src/MARIGOLD/iate.py:342]: There's also calc_vgj

Comment (ajdix) [reply src/MARIGOLD/iate.py:342]: But in general, could implement other correlations for vgj

**Review by ajdix — APPROVED** [review]

File: src/MARIGOLD/iate.py (hunk 1 above)
Comment (felizdavidad) [reply src/MARIGOLD/iate.py:46]: Resolved

File: src/MARIGOLD/iate.py (hunk 2 above)
Comment (felizdavidad) [reply src/MARIGOLD/iate.py:63]: Resolved

File: src/MARIGOLD/iate.py (hunk 5 above)
Comment (felizdavidad) [reply src/MARIGOLD/iate.py:228]: I'm not so sure; this calculation occurs now during the looping part of the script as it marches along the z-axis

I think cond.calc_cd, cond.calc_mu_eff, etc. use local probe data which we don't have access to as we depart from the initial boundary condition --

They'd need to take variable area-averaged or void-area-averaged parameters as inputs, but I don't think that's really what these functions are made for

Less predicting CD or mu_eff at specific L/Ds, more finding them at whatever port we have data?

Correct me if I'm wrong

**Pull request merged**